### PR TITLE
[15007] Add dll support to Dynamic Types API

### PR DESCRIPTION
**Pull request by jparisu**
Signed-off-by: jparisu <javierparis@eprosima.com>

## Description

When used in plotjuggler fastdds plugin : https://github.com/eProsima/plotjuggler-fastdds-plugins
there has been a problem while using the Dynamic Types API in Windows because some methods does not have the dll support.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [ ] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [ ] Any new/modified methods have been properly documented using Doxygen. 
- [ ] Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [x] Documentation builds and tests pass locally. 
- [ ] New feature has been added to the `versions.md` file (if applicable).
- [ ] New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by MiguelCompany — DISMISSED** [review]
LGTM, but please fix uncrustify

File: include/fastrtps/types/TypeDescriptor.h [hunk 1]
```diff
@@ -63,97 +63,97 @@ class TypeDescriptor
...
 
-    void annotation_set_key(
-            bool key);
+    RTPS_DllAPI void annotation_set_key(
+            RTPS_DllAPI bool key);
 
-    void annotation_set_non_serialized(
-            bool non_serialized);
+    RTPS_DllAPI void annotation_set_non_serialized(
+            RTPS_DllAPI bool non_serialized);
```
Comment (jsantiago-eProsima) [inline include/fastrtps/types/TypeDescriptor.h:156]:
```suggestion
            bool non_serialized);
```

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

**Comment by jparisu** [comment]
Waiting for PR https://github.com/eProsima/plotjuggler-fastdds-plugins/pull/10 to check this branch fixes the problem

**Comment by jparisu** [comment]
Ready to merge :+1:

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Review by jsantiago-eProsima — APPROVED** [review]

**Pull request merged**